### PR TITLE
Restore stream in/out timeouts to a sane value

### PR DESCRIPTION
**Pull request by ggezer**
Earlier default (1hour) is an abnormal value for streaming timeout for a single chunk.
Let's make it more saner and watch if it causes any issues.

### CODE REVIEW

File: core/common/src/main/java/alluxio/conf/PropertyKey.java [hunk 1]
```diff
@@ -4853,7 +4853,7 @@ public String toString() {
       new Builder(Name.USER_STREAMING_DATA_READ_TIMEOUT)
           .setAlias("alluxio.user.network.data.timeout.ms", Name.USER_NETWORK_DATA_TIMEOUT,
               Name.USER_STREAMING_DATA_TIMEOUT)
-          .setDefaultValue("1h")
+          .setDefaultValue("1m")
```
Comment (yuzhu) [inline core/common/src/main/java/alluxio/conf/PropertyKey.java:4856]: can we be a little more conservative? I would propose 5m? users have very different ufs set ups.

**Comment by jiacheliu3** [comment]
This property was changed from 30s to 1h in https://github.com/Alluxio/alluxio/pull/12463 because when the worker is extremely busy or the UFS is slow, the response time can be spurious. 

I don't know too much about the exact pros/cons of this having a very large value. But AFAIR the client data path does not have much retry logic embedded so if the read times out, we rely on the compute logic to retry? If that is true, then I guess the root question becomes: If the worker/UFS is really slow, do we prefer the query to fail early(and make the admin cry) or try very hard(risk the system stress). WDYT @apc999 @ggezer @yuzhu @beinan @rongrong  ?

File: core/common/src/main/java/alluxio/conf/PropertyKey.java [hunk 2]
```diff
@@ -4853,7 +4853,7 @@ public String toString() {
       new Builder(Name.USER_STREAMING_DATA_READ_TIMEOUT)
           .setAlias("alluxio.user.network.data.timeout.ms", Name.USER_NETWORK_DATA_TIMEOUT,
               Name.USER_STREAMING_DATA_TIMEOUT)
-          .setDefaultValue("1h")
+          .setDefaultValue("5m")
```
Comment (jiacheliu3) [inline core/common/src/main/java/alluxio/conf/PropertyKey.java:4856]: I guess the "problem" behind this property is, it doesn't really timeout easily in our existing tests. So when there are problems, they most likely take place in a very large production workload. So maybe it's hard to just update this value and try our tests. A potentially better way is to rely on people's experience (yea not too reliable, i know).

Comment (ggezer) [reply core/common/src/main/java/alluxio/conf/PropertyKey.java]: I don't think there's any compute app out there that would tolerate 1h wait time. They usually timeout requests between 5-8 mins. So our defaults should be reflective of that.

After considering more, I set this to 3m and I believe that should be the most as the default value. If there are clusters out there that have very slow connections, the way should be tuning them individually not via bloated defaults like we have here.

**Comment by jiacheliu3** [comment]
> This property was changed from 30s to 1h in #12463 because when the worker is extremely busy or the UFS is slow, the response time can be spurious.
> 
> (Correct me if i'm wrong, i'm not so familiar with the grpc streaming code part) In the context of this PR, this specific timeout is for sending/receiving one single data chunk in a stream (a little like the timeout between two messages but not 100%). It does look weird that we allow such a huge timeout between messages. But if this timeout exception is thrown, looks to me that the `GrpcDataReader` is closed and the read stream will fail? I'm not sure which one is better:
> 
> 1. Having one huge timeout with 1 try and bet things will work eventually
> 2. Having small timeout with more retries and hope the worst-case response time is not always bad
> 
> Maybe it makes sense to retry only when the error is retriable like a timeout, not something like a closed connection?
> 
> I don't know too much about the exact pros/cons of this having a very large value here. But AFAIR the client data path does not have much retry logic embedded so if the read times out, we rely on the compute logic to retry? If that is true, then I guess the root question becomes: **If the worker/UFS is really slow, do we prefer the query to fail early(and make the admin cry) or try very hard(risk the system stress)**.
> 
> WDYT @apc999 @ggezer @yuzhu @beinan @rongrong ?

Even though this discussion can be off-the-topic for this PR, I still figure this can be a meaningful discussion. I feel like we typically use timeouts and retries ad-hoc, meaning we throw a configurable timeout/retry to guard particular function calls or just because the method requires a timeout param. The question "how long does an alluxio I/O wait before it times out?" is rarely answered in a systematic way.

**Comment by ggezer** [comment]
alluxio-bot, merge this please

**Pull request merged**